### DISCUSSION TITLE
Invalid command name "140458435327488check_dpi_scaling" error

### DISCUSSION THREAD

**Discussion by nimanthadilz** [TomSchimansky/CustomTkinter#607] [Q&A] [answered]
I am developing a simple data visualization tool using CustomTkinter (https://github.com/pycharts/pycharts).

The program runs smoothly. After doing some operations, when I close the program, I get an error in the terminal and the process seems frozen. The error is shown below.
```
invalid command name "140053293976576check_dpi_scaling"
    while executing
"140053293976576check_dpi_scaling"
    ("after" script)
invalid command name "140053294126592update"
    while executing
"140053294126592update"
    ("after" script)
```
I experienced this on Ubuntu 22.04.1 LTS.
I have two collaborators working on the same project who use Windows 11 and faced the same error.

When we convert our source code into an executable, even though the user doesn't see the error, the process is hanging out there after closing the program.

If anyone knows how to fix this, please help me to fix this.

**Comment by Akascape** [comment, marked as answer]
@nimanthadilz Are you using more than one mainloop in the program?

**Reply by nimanthadilz** [reply]
I think I am not.
```python
if __name__=="__main__":
    # root = tk.Tk()
    root = customtkinter.CTk()
    root.minsize(1480, 720)
    App(root)
    root.mainloop()
```

This is where mainloop is called.

**Reply by Akascape** [reply]
@nimanthadilz Try adding 
 ```python
self.protocol("WM_DELETE_WINDOW", self.on_closing) 

def on_closing(self, event=0): 
         self.destroy()
```
Or
```python
import sys
self.protocol("WM_DELETE_WINDOW", self.on_closing) 

def on_closing(self, event=0): 
         sys.exit()
```

This will force quit the processes.

**Reply by nimanthadilz** [reply]
I tried your suggestion. I changed my code to the one below.
```python
if __name__=="__main__":
    root = customtkinter.CTk()
    root.protocol("WM_DELETE_WINDOW", sys.exit)
    root.minsize(1480, 720)
    App(root)
    root.mainloop()
```

Now I don't experience that previous error. I checked with my collaborators using Windows. They also confirmed that this fix works.

Thank you for your support. It really helped.
